File: NetworkMonitorApi/ClientApp/src/app/models/log.ts
import { LogType } from '../enums';

export class Log {

  userId: string;
  logType: LogType;
  source: string;
  message: string;
  dateCreated: string;

  constructor(userId: string, logType: LogType, source: string, message: string, dateCreated: string) {

    if (!(this instanceof Log)) {
      return new Log(userId, logType, source, message, dateCreated);
    }

    this.userId = userId;
    this.logType = logType;
    this.message = message;
    this.source = source;
  }
  
}

File: NetworkMonitorApi/Models/Log.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;
using static NetworkMonitorApi.CustomEnums;

namespace NetworkMonitorApi.Models
{
    public class Log
    {
        [Key]
        public int LogId { get; set; }
        public LogType LogType { get; set; }
        public string Message { get; set; }
        public string UserId { get; set; }
        public string Source { get; set; }
        public DateTime DateCreated { get; set; }

        public bool Checked { get; set; }
    }
}
